src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,106 @@
+            statement.setBigDecimal(2, product.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                product.setId(resultSet.getLong(1));
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create product - " + product, e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        "SELECT * FROM products WHERE deleted = 0 AND product_id = ?")) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                return Optional.of(extractProductFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get product from DB with ID = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        "SELECT * FROM products WHERE deleted = false")) {
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                Product product = extractProductFromResultSet(rs);
+                products.add(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get items from DB", e);
+        }
+        return products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(
+                        "UPDATE products SET name = ?, price = ? WHERE product_id = ?")) {
+            statement.setString(1, product.getName());
+            statement.setString(2, String.valueOf(product.getPrice()));
+            statement.setString(3, String.valueOf(product.getId()));
+            statement.executeUpdate();
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update product - " + product, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        "UPDATE products SET deleted = 1 WHERE product_id = ?")) {
+            statement.setString(1, String.valueOf(id));
+            int i = statement.executeUpdate();
+            return i == 1;
```

Comment:
```suggestion
            return statement.executeUpdate();
```

Reply:
executeUpdate returns int. The method returns boolean

Reply:
I have changed to return staement.executeUpdate() == 1;